packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -0,0 +1,631 @@
+    buffer._putUint8(_strokePaintTag);
+    buffer._putUint32(color);
+    buffer._putUint8(blendMode);
+    buffer._putInt32(paintId);
+    return paintId;
+  }
+
+  /// Encode a paint object in the current buffer, returning the identifier
+  /// assigned to it.
+  ///
+  /// [color] is the 32-bit ARBG color representation used by Flutter
+  /// internally. The [strokeCap], [strokeJoin], [blendMode], [style]
+  /// fields should be the index of the corresponding enumeration.
+  ///
+  /// This method is only used to write the paint used for fill commands.
+  /// To write a paint used for a stroke command, see [writePaintStroke].
+  int writePaintStroke(
+    VectorGraphicsBuffer buffer,
+    int color,
+    int strokeCap,
+    int strokeJoin,
+    int blendMode,
+    double strokeMiterLimit,
+    double strokeWidth,
+  ) {
+    if (buffer._decodePhase.index > _DecodePhase.paints.index) {
+      throw StateError('Paints must be encoded together.');
+    }
+    buffer._decodePhase = _DecodePhase.paints;
+    final int paintId = buffer._nextPaintId++;
+    buffer._putUint8(_strokePaintTag);
+    buffer._putUint32(color);
+    buffer._putUint8(strokeCap);
+    buffer._putUint8(strokeJoin);
+    buffer._putUint8(blendMode);
+    buffer._putFloat64(strokeMiterLimit);
+    buffer._putFloat64(strokeWidth);
+    buffer._putInt32(paintId);
+    return paintId;
+  }
+
+  void _readFillPaint(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int color = buffer.getUint32();
+    final int blendMode = buffer.getUint8();
+    final int id = buffer.getInt32();
+
+    listener?.onPaintObject(
+      color,
+      null,
+      null,
+      blendMode,
+      null,
+      null,
+      0, // Fill
+      id,
+    );
+  }
+
+  void _readStrokePaint(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int color = buffer.getUint32();
+    final int strokeCap = buffer.getUint8();
+    final int strokeJoin = buffer.getUint8();
+    final int blendMode = buffer.getUint8();
+    final double strokeMiterLimit = buffer.getFloat64();
+    final double strokeWidth = buffer.getFloat64();
+    final int id = buffer.getInt32();
+
+    listener?.onPaintObject(
+      color,
+      strokeCap,
+      strokeJoin,
+      blendMode,
+      strokeMiterLimit,
+      strokeWidth,
+      1, // Stroke
+      id,
+    );
+  }
+
+  /// Begin writing a new path to the [buffer], returing the identifier
+  /// assigned to it.
+  ///
+  /// The [fillType] argument is either `1` for a fill or `0` for a stroke.
+  ///
+  /// Throws a [StateError] if there is already an active path.
+  int writeStartPath(VectorGraphicsBuffer buffer, int fillType) {
+    if (buffer._currentPathId != -1) {
+      throw StateError('There is already an active Path.');
+    }
+    if (buffer._decodePhase.index > _DecodePhase.paths.index) {
+      throw StateError('Paths must be encoded together');
+    }
+    buffer._decodePhase = _DecodePhase.paths;
```

Comment:
here and elsewhere: it seems confusing to use DecoePhase for both encoding and decoding...

Reply:
Done